src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2826,13 +2826,24 @@ private uint getJitFlags(ref CORJIT_FLAGS flags, uint sizeInBytes)
 #endif
                )
             {
+#if !READYTORUN
                 // This list needs to match the list of intrinsics we can generate detection code for
                 // in HardwareIntrinsicHelpers.EmitIsSupportedIL.
+#else
+                // For ReadyToRun, this list needs to match up with the behavior of FilterNamedIntrinsicMethodAttribs
+                // In particular, that this list of supported hardware will not generate non-SSE2 safe instruction
+                // sequences when paired with the behavior in FilterNamedIntrinsicMethodAttribs
+#endif
                 flags.Set(CorJitFlag.CORJIT_FLAG_USE_AES);
                 flags.Set(CorJitFlag.CORJIT_FLAG_USE_PCLMULQDQ);
                 flags.Set(CorJitFlag.CORJIT_FLAG_USE_SSE3);
                 flags.Set(CorJitFlag.CORJIT_FLAG_USE_SSSE3);
                 flags.Set(CorJitFlag.CORJIT_FLAG_USE_LZCNT);
+#if READYTORUN
```

Comment:
Why are these 3 supported under Ready2Run but not under other scenarios? Is the other path here for Crossgen1/2?

Reply:
The other path is for CoreRT, which isn't a crossgen scenario at all. (And really shouldn't be in this repo, but the current amount of CoreRT logic embedded in here isn't really large.)

Reply:
It being for CoreRT was where I was getting confused. Thanks for clarifying!